### ISSUE TITLE
Improve description of partial derivatives

### ISSUE THREAD

**Issue by HSchatzTLK** [modelica/fmi-guides#26]
FMI Standard, 2.2.12. Getting Partial Derivatives:
- definition of adjoint derivatives is prone to misunderstandings because the seed and sensitivity vectors are not just tranposed but have different dimensions

Implementer's Guide, 9.1. Partial Derivatives:
- the examples are hard to follow without the basic definitions of directional and adjoint derivatives
- Add example for matrix-free Jacobian-vector products to directional derivatives, e.g. for Krylov methods
- Adjoint derivatives should be chapter 9.1.2 instead of 9.1.1.1 (?)

**Comment by pmai** [comment]
I have fixed the heading level for adjoint derivatives. For the other suggested improvements to the IG I would welcome a PR in this direction.

**Comment by HSchatzTLK** [comment]
Ok, I'll try to have an PR ready by the end of the week

**Comment by pmai** [comment]
Thanks for the proposed changes, have been merged.
